lib/tetra/facades/bash.rb

```diff
@@ -31,12 +31,17 @@ def bash
           bashrc_file.write(bashrc_content)
           bashrc_file.flush
 
-          run_interactive("bash --rcfile #{bashrc_file.path} -i")
-          history = File.read(history_file)
-          log.debug "history contents:"
-          log.debug history
+          if command
+            run("bash --rcfile #{bashrc_file.path} -i -c '#{command}'")
+            [command]
```

Comment:
The only potential issue I see is that the history returned by `run` is subsequently used to generate the `build.sh` file - tetra plays with `ant` and `mvn` options (in future, other build systems might get supported) in order to make sure they can run without network access at RPM building time.

So I wonder if it would make more sense to pass a filename to `--script` instead, and split it to lines here.

Can your use case be fulfilled with just a one-liner?

Reply:
In our case we do a

```
tetra dry-run -c 'cd src/jruby-9.1.15; mvn -P \!truffle -Djruby.default.ruby.version=2.3'
```

so it's two commands in one line. A script file would also work for us but make it slightly more complex.

I can't really judge the side effects that this might have right now, so it's your call :)

Reply:
https://github.com/moio/tetra/pull/20 should fix this issue.